src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioErrorReportingService.ExceptionFormatting.cs

```diff
@@ -0,0 +1,210 @@
+        private const string AsyncMethodPrefix = "async ";
+
+        private static string GetFormattedExceptionStack(Exception exception)
+        {
+            var aggregate = exception as AggregateException;
+            if (aggregate != null)
+            {
+                return GetStackForAggregateException(exception, aggregate);
+            }
+
+            return GetStackForException(exception, false);
+        }
+
+        private static string GetStackForAggregateException(Exception exception, AggregateException aggregate)
+        {
+            var text = GetStackForException(exception, true);
+            for (int i = 0; i < aggregate.InnerExceptions.Count; i++)
+            {
+                text = string.Format(ServicesVSResources._0_1_InnerException_2_3_4_5, text,
+                    Environment.NewLine, i, GetFormattedExceptionStack(aggregate.InnerExceptions[i]), "<---", Environment.NewLine);
+            }
+
+            return text;
+        }
+
+        private static string GetStackForException(Exception exception, bool includeMessageOnly)
+        {
+            var message = exception.Message;
+            var className = exception.GetType().ToString();
+            var stackText = message.Length <= 0
+                ? className
+                : className + " : " + message;
+            var innerException = exception.InnerException;
+            if (innerException != null)
+            {
+                if (includeMessageOnly)
+                {
+                    do
+                    {
+                        stackText += " ---> " + innerException.Message;
+                        innerException = innerException.InnerException;
+                    } while (innerException != null);
+                }
+                else
+                {
+                    stackText += " ---> " + GetFormattedExceptionStack(innerException) + Environment.NewLine +
+                            "   " + ServicesVSResources.End_of_inner_exception_stack;
+                }
+            }
+            
+            return stackText + Environment.NewLine + GetAsyncStackTrace(exception);
+        }
+
+        private static string GetAsyncStackTrace(Exception exception)
+        {
+            var stackTrace = new StackTrace(exception, true);
+
+            var stackFrames = stackTrace.GetFrames();
+            if (stackFrames == null)
+            {
+                return string.Empty;
+            }
+
+            var firstFrame = true;
+            var stringBuilder = new StringBuilder(255);
+
+            foreach (var frame in stackFrames)
+            {
+                var method = frame.GetMethod();
+                if (method == null)
+                {
+                    continue;
+                }
+
+                var declaringType = method.DeclaringType;
+                if (declaringType != null && typeof(INotifyCompletion).IsAssignableFrom(declaringType))
+                {
+                    continue;
```

Comment:
so it looks like you're doing this to skip over poitnless methods (cool).  However, i would only skip if you're certain it's safe to do so.  For example, if method == null above, you may not want to skip it...